PCem errors out when compile the disc part in EndeavourOS

hello. i was trying to compile PCem on my new EndeavourOS install, just because i wanted to set up a machine that i could TAS, and for some reason, it gave me an error about "incompatible pointers" and "too many arguments":

```
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT     -O3 -fcommon -MT pcem-disc.o -MD -MP -MF .deps/pcem-disc.Tpo -c -o pcem-disc.o `test -f 'disc.c' || echo './'`disc.c
disc.c: In function ‘disc_reset’:
disc.c:262:37: error: passing argument 2 of ‘timer_add’ from incompatible pointer type [-Wincompatible-pointer-types]
  262 |         timer_add(&disc_poll_timer, disc_poll, NULL, 0);
      |                                     ^~~~~~~~~
      |                                     |
      |                                     void (*)(void)
In file included from ibm.h:5,
                 from disc.c:1:
timer.h:47:42: note: expected ‘void (*)(void *)’ but argument is of type ‘void (*)(void)’
   47 | void timer_add(pc_timer_t *timer, void (*callback)(void *p), void *p, int start_timer);
      |                                   ~~~~~~~^~~~~~~~~~~~~~~~~~
disc.c:188:6: note: ‘disc_poll’ declared here
  188 | void disc_poll()
      |      ^~~~~~~~~
disc.c: In function ‘disc_stop’:
disc.c:339:17: error: too many arguments to function ‘drives[drive].stop’; expected 0, have 1
  339 |                 drives[drive].stop(drive);
      |                 ^~~~~~             ~~~~~
In file included from disc.c:4:
disc.h:9:16: note: declared here
    9 |         void (*stop)();
      |                ^~~~
make[1]: *** [Makefile:2451: pcem-disc.o] Error 1
make[1]: Leaving directory '/home/dominic/pcem/src'
make: *** [Makefile:373: all-recursive] Error 1
```


![Image](https://github.com/user-attachments/assets/c8d19138-8cd0-43a8-ae02-89993a7f2844) 
heres proof i run EndeavourOS as well

i don't know if this is just an issue on EndeavourOS or i might just not have appropriate packages, but i honestly hope i can get help, because i don't know what's going on.

```
Making all in src
make[1]: Entering directory '/home/dominic/pcem/src'
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT     -O3 -fcommon -MT pcem-disc.o -MD -MP -MF .deps/pcem-disc.Tpo -c -o pcem-disc.o `test -f 'disc.c' || echo './'`disc.c
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT     -O3 -fcommon -MT pcem-disc_img.o -MD -MP -MF .deps/pcem-disc_img.Tpo -c -o pcem-disc_img.o `test -f 'disc_img.c' || echo './'`disc_img.c
disc.c:188:6: error: conflicting types for ‘disc_poll’; have ‘void(void *)’
  188 | void disc_poll(void*)
      |      ^~~~~~~~~
In file included from disc.c:4:
disc.h:22:6: note: previous declaration of ‘disc_poll’ with type ‘void(void)’
   22 | void disc_poll();
      |      ^~~~~~~~~
disc.c: In function ‘disc_stop’:
disc.c:339:17: error: too many arguments to function ‘drives[drive].stop’; expected 0, have 1
  339 |                 drives[drive].stop(drive);
      |                 ^~~~~~             ~~~~~
disc.h:9:16: note: declared here
    9 |         void (*stop)();
      |                ^~~~
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT     -O3 -fcommon -MT pcem-disc_sector.o -MD -MP -MF .deps/pcem-disc_sector.Tpo -c -o pcem-disc_sector.o `test -f 'disc_sector.c' || echo './'`disc_sector.c
make[1]: *** [Makefile:2451: pcem-disc.o] Error 1
make[1]: *** Waiting for unfinished jobs....
mv -f .deps/pcem-disc_sector.Tpo .deps/pcem-disc_sector.Po
mv -f .deps/pcem-disc_img.Tpo .deps/pcem-disc_img.Po
make[1]: Leaving directory '/home/dominic/pcem/src'
make: *** [Makefile:373: all-recursive] Error 1
```

I'm still getting errors.

```
Making all in src
make[1]: Entering directory '/home/dominic/pcem/src'
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT     -O3 -fcommon -MT pcem-disc.o -MD -MP -MF .deps/pcem-disc.Tpo -c -o pcem-disc.o `test -f 'disc.c' || echo './'`disc.c
disc.c: In function ‘disc_stop’:
disc.c:339:17: error: too many arguments to function ‘drives[drive].stop’; expected 0, have 1
  339 |                 drives[drive].stop(drive);
      |                 ^~~~~~             ~~~~~
In file included from disc.c:4:
disc.h:9:16: note: declared here
    9 |         void (*stop)();
      |                ^~~~
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT     -O3 -fcommon -MT pcem-disc_fdi.o -MD -MP -MF .deps/pcem-disc_fdi.Tpo -c -o pcem-disc_fdi.o `test -f 'disc_fdi.c' || echo './'`disc_fdi.c
make[1]: *** [Makefile:2451: pcem-disc.o] Error 1
make[1]: *** Waiting for unfinished jobs....
mv -f .deps/pcem-disc_fdi.Tpo .deps/pcem-disc_fdi.Po
make[1]: Leaving directory '/home/dominic/pcem/src'
make: *** [Makefile:373: all-recursive] Error 1
```
Looks like we're getting somewhere.

```
Making all in src
make[1]: Entering directory '/home/dominic/pcem/src'
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT     -O3 -fcommon -MT pcem-disc.o -MD -MP -MF .deps/pcem-disc.Tpo -c -o pcem-disc.o `test -f 'disc.c' || echo './'`disc.c
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT     -O3 -fcommon -MT pcem-disc_img.o -MD -MP -MF .deps/pcem-disc_img.Tpo -c -o pcem-disc_img.o `test -f 'disc_img.c' || echo './'`disc_img.c
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT     -O3 -fcommon -MT pcem-disc_sector.o -MD -MP -MF .deps/pcem-disc_sector.Tpo -c -o pcem-disc_sector.o `test -f 'disc_sector.c' || echo './'`disc_sector.c
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT     -O3 -fcommon -MT pcem-dma.o -MD -MP -MF .deps/pcem-dma.Tpo -c -o pcem-dma.o `test -f 'dma.c' || echo './'`dma.c
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT     -O3 -fcommon -MT pcem-esdi_at.o -MD -MP -MF .deps/pcem-esdi_at.Tpo -c -o pcem-esdi_at.o `test -f 'esdi_at.c' || echo './'`esdi_at.c
esdi_at.c:3:9: warning: ‘_GNU_SOURCE’ redefined
    3 | #define _GNU_SOURCE
      |         ^~~~~~~~~~~
<command-line>: note: this is the location of the previous definition
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT     -O3 -fcommon -MT pcem-f82c710_upc.o -MD -MP -MF .deps/pcem-f82c710_upc.Tpo -c -o pcem-f82c710_upc.o `test -f 'f82c710_upc.c' || echo './'`f82c710_upc.c
mv -f .deps/pcem-disc.Tpo .deps/pcem-disc.Po
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT     -O3 -fcommon -MT pcem-fdc.o -MD -MP -MF .deps/pcem-fdc.Tpo -c -o pcem-fdc.o `test -f 'fdc.c' || echo './'`fdc.c
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT     -O3 -fcommon -MT pcem-fdc37c665.o -MD -MP -MF .deps/pcem-fdc37c665.Tpo -c -o pcem-fdc37c665.o `test -f 'fdc37c665.c' || echo './'`fdc37c665.c
fdc.c: In function ‘fdc_init’:
fdc.c:1360:31: error: passing argument 2 of ‘timer_add’ from incompatible pointer type [-Wincompatible-pointer-types]
 1360 |         timer_add(&fdc.timer, fdc_callback, NULL, 0);
      |                               ^~~~~~~~~~~~
      |                               |
      |                               void (*)(void)
In file included from ibm.h:5,
                 from fdc.c:3:
timer.h:47:42: note: expected ‘void (*)(void *)’ but argument is of type ‘void (*)(void)’
   47 | void timer_add(pc_timer_t *timer, void (*callback)(void *p), void *p, int start_timer);
      |                                   ~~~~~~~^~~~~~~~~~~~~~~~~~
fdc.c:801:6: note: ‘fdc_callback’ declared here
  801 | void fdc_callback()
      |      ^~~~~~~~~~~~
mv -f .deps/pcem-disc_img.Tpo .deps/pcem-disc_img.Po
make[1]: *** [Makefile:2549: pcem-fdc.o] Error 1
make[1]: *** Waiting for unfinished jobs....
mv -f .deps/pcem-disc_sector.Tpo .deps/pcem-disc_sector.Po
mv -f .deps/pcem-fdc37c665.Tpo .deps/pcem-fdc37c665.Po
mv -f .deps/pcem-f82c710_upc.Tpo .deps/pcem-f82c710_upc.Po
mv -f .deps/pcem-esdi_at.Tpo .deps/pcem-esdi_at.Po
mv -f .deps/pcem-dma.Tpo .deps/pcem-dma.Po
make[1]: Leaving directory '/home/dominic/pcem/src'
make: *** [Makefile:373: all-recursive] Error 1
```
I also did what it told me to do at line 339 of disc.c and I'm getting new errors this time

```
gcc -std=c11 -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT    -DRELEASE_BUILD -O3 -fcommon -MT pcem-wx-sdl2-midi.o -MD -MP -MF .deps/pcem-wx-sdl2-midi.Tpo -c -o pcem-wx-sdl2-midi.o `test -f 'wx-sdl2-midi.c' || echo './'`wx-sdl2-midi.c
mv -f .deps/pcem-wx-sdl2-midi.Tpo .deps/pcem-wx-sdl2-midi.Po
gcc -std=c11 -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT    -DRELEASE_BUILD -O3 -fcommon -MT pcem-codegen_backend_x86-64.o -MD -MP -MF .deps/pcem-codegen_backend_x86-64.Tpo -c -o pcem-codegen_backend_x86-64.o `test -f 'codegen_backend_x86-64.c' || echo './'`codegen_backend_x86-64.c
gcc -std=c11 -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT    -DRELEASE_BUILD -O3 -fcommon -MT pcem-codegen_backend_x86-64_ops.o -MD -MP -MF .deps/pcem-codegen_backend_x86-64_ops.Tpo -c -o pcem-codegen_backend_x86-64_ops.o `test -f 'codegen_backend_x86-64_ops.c' || echo './'`codegen_backend_x86-64_ops.c
codegen_backend_x86-64.c: In function ‘codegen_backend_init’:
codegen_backend_x86-64.c:335:9: error: implicit declaration of function ‘asm’ [-Wimplicit-function-declaration]
  335 |         asm(
      |         ^~~
codegen_backend_x86-64.c:336:31: error: expected ‘)’ before ‘:’ token
  336 |                 "stmxcsr %0\n"
      |                               ^
      |                               )
  337 |                 : "=m" (cpu_state.old_fp_control)
      |                 ~              
codegen_backend_x86-64.c:335:12: note: to match this ‘(’
  335 |         asm(
      |            ^
make[1]: *** [Makefile:5601: pcem-codegen_backend_x86-64.o] Error 1
make[1]: *** Waiting for unfinished jobs....
mv -f .deps/pcem-wx-config.Tpo .deps/pcem-wx-config.Po
mv -f .deps/pcem-x86seg.Tpo .deps/pcem-x86seg.Po
mv -f .deps/pcem-wx-sdl2-video-gl3.Tpo .deps/pcem-wx-sdl2-video-gl3.Po
mv -f .deps/pcem-wx-dialogbox.Tpo .deps/pcem-wx-dialogbox.Po
mv -f .deps/pcem-wx-app.Tpo .deps/pcem-wx-app.Po
mv -f .deps/pcem-wx-deviceconfig.Tpo .deps/pcem-wx-deviceconfig.Po
mv -f .deps/pcem-wx-resources.Tpo .deps/pcem-wx-resources.Po
mv -f .deps/pcem-wx-main.Tpo .deps/pcem-wx-main.Po
mv -f .deps/pcem-codegen_backend_x86-64_ops.Tpo .deps/pcem-codegen_backend_x86-64_ops.Po
mv -f .deps/pcem-wx-status.Tpo .deps/pcem-wx-status.Po
mv -f .deps/pcem-wx-shaderconfig.Tpo .deps/pcem-wx-shaderconfig.Po
mv -f .deps/pcem-wx-createdisc.Tpo .deps/pcem-wx-createdisc.Po
mv -f .deps/pcem-wx-joystickconfig.Tpo .deps/pcem-wx-joystickconfig.Po
mv -f .deps/pcem-wx-utils.Tpo .deps/pcem-wx-utils.Po
mv -f .deps/pcem-vid_voodoo_render.Tpo .deps/pcem-vid_voodoo_render.Po
make[1]: Leaving directory '/home/dominic/pcem/src'
make: *** [Makefile:373: all-recursive] Error 1
```
another error

> [@TheVibster](https://github.com/TheVibster) please check out that pull request linked right above, see if it builds fine for you.

```
gcc -DPACKAGE_NAME=\"PCem\" -DPACKAGE_TARNAME=\"pcem\" -DPACKAGE_VERSION=\"v17\" -DPACKAGE_STRING=\"PCem\ v17\" -DPACKAGE_BUGREPORT=\"Sarah\ Walker\ \<pcem@pcem-emulator.co.uk\>\" -DPACKAGE_URL=\"\" -DPACKAGE=\"pcem\" -DVERSION=\"v17\" -DHAVE_LIBPTHREAD=1 -DHAVE_LIBGL=1 -DHAVE_LIBOPENAL=1 -I.    -I/usr/lib/wx/include/gtk3-unicode-3.2 -I/usr/include/wx-3.2 -DWXUSINGDLL -D__WXGTK3__ -D__WXGTK__ -D_FILE_OFFSET_BITS=64 -I/usr/include/SDL2 -D_GNU_SOURCE=1 -D_REENTRANT    -DRELEASE_BUILD -O3 -fcommon -MT pcem-disc_fdi.o -MD -MP -MF .deps/pcem-disc_fdi.Tpo -c -o pcem-disc_fdi.o `test -f 'disc_fdi.c' || echo './'`disc_fdi.c
disc.c: In function ‘disc_reset’:
disc.c:262:37: error: passing argument 2 of ‘timer_add’ from incompatible pointer type [-Wincompatible-pointer-types]
  262 |         timer_add(&disc_poll_timer, disc_poll, NULL, 0);
      |                                     ^~~~~~~~~
      |                                     |
      |                                     void (*)(void)
In file included from ibm.h:5,
                 from disc.c:1:
timer.h:47:42: note: expected ‘void (*)(void *)’ but argument is of type ‘void (*)(void)’
   47 | void timer_add(pc_timer_t *timer, void (*callback)(void *p), void *p, int start_timer);
      |                                   ~~~~~~~^~~~~~~~~~~~~~~~~~
disc.c:188:6: note: ‘disc_poll’ declared here
  188 | void disc_poll()
      |      ^~~~~~~~~
disc.c: In function ‘disc_stop’:
disc.c:339:17: error: too many arguments to function ‘drives[drive].stop’; expected 0, have 1
  339 |                 drives[drive].stop(drive);
      |                 ^~~~~~             ~~~~~
In file included from disc.c:4:
disc.h:9:16: note: declared here
    9 |         void (*stop)();
      |                ^~~~
make[1]: *** [Makefile:2451: pcem-disc.o] Error 1
make[1]: *** Waiting for unfinished jobs....
mv -f .deps/pcem-device.Tpo .deps/pcem-device.Po
mv -f .deps/pcem-codegen_ops_jump.Tpo .deps/pcem-codegen_ops_jump.Po
mv -f .deps/pcem-cpu.Tpo .deps/pcem-cpu.Po
mv -f .deps/pcem-disc_fdi.Tpo .deps/pcem-disc_fdi.Po
mv -f .deps/pcem-codegen_ops_stack.Tpo .deps/pcem-codegen_ops_stack.Po
mv -f .deps/pcem-codegen_ops_shift.Tpo .deps/pcem-codegen_ops_shift.Po
mv -f .deps/pcem-codegen_ops_logic.Tpo .deps/pcem-codegen_ops_logic.Po
mv -f .deps/pcem-codegen_ops_misc.Tpo .deps/pcem-codegen_ops_misc.Po
mv -f .deps/pcem-codegen_ops_mov.Tpo .deps/pcem-codegen_ops_mov.Po
mv -f .deps/pcem-808x.Tpo .deps/pcem-808x.Po
mv -f .deps/pcem-386_dynarec_ops.Tpo .deps/pcem-386_dynarec_ops.Po
mv -f .deps/pcem-386_dynarec.Tpo .deps/pcem-386_dynarec.Po
make[1]: Leaving directory '/home/dominic/pcem/src'
make: *** [Makefile:373: all-recursive] Error 1
```
now im getting different errors.

--- a/src/disc.c
+++ b/src/disc.c
@@ -185,7 +185,7 @@ int disc_hole(int drive)
 	}
 }
 
-void disc_poll()
+void disc_poll(void*)
 {
         timer_advance_u64(&disc_poll_timer, disc_period * TIMER_USEC);
 

--- a/src/disc.h
+++ b/src/disc.h
@@ -6,7 +6,7 @@ typedef struct
         void (*readaddress)(int drive, int track, int side, int density);
         void (*format)(int drive, int track, int side, int density, uint8_t fill);
         int (*hole)(int drive);
-        void (*stop)();
+        void (*stop)(int drive);
         void (*poll)();
 } DRIVE;
 
@@ -19,7 +19,7 @@ void disc_new(int drive, char *fn);
 void disc_close(int drive);
 void disc_init();
 void disc_reset();
-void disc_poll();
+void disc_poll(void*);
 void disc_seek(int drive, int track);
 void disc_readsector(int drive, int sector, int track, int side, int density, int sector_size);
 void disc_writesector(int drive, int sector, int track, int side, int density, int sector_size);
@@ -33,7 +33,7 @@ void disc_set_drivesel(int drive);
 void disc_set_motor_enable(int motor_enable);
 extern int disc_drivesel;
 
-void fdc_callback();
+void fdc_callback(void*);
 int  fdc_data(uint8_t dat);
 void fdc_spindown();
 void fdc_finishread();

--- a/src/disc_fdi.c
+++ b/src/disc_fdi.c
@@ -260,7 +260,7 @@ static uint8_t decodefm(uint16_t dat)
         return temp;
 }
 
-void fdi_stop()
+void fdi_stop(int)
 {
 //        pclog("fdi_stop\n");
         fdi_inread = fdi_inwrite = fdi_inreadaddr = 0;

--- a/src/disc_fdi.h
+++ b/src/disc_fdi.h
@@ -7,5 +7,5 @@ void fdi_writesector(int drive, int sector, int track, int side, int density, in
 void fdi_readaddress(int drive, int sector, int side, int density);
 void fdi_format(int drive, int sector, int side, int density, uint8_t fill);
 int fdi_hole(int drive);
-void fdi_stop();
+void fdi_stop(int drive);
 void fdi_poll();

--- a/src/fdc.c
+++ b/src/fdc.c
@@ -80,7 +80,7 @@ typedef struct FDC
 
 static FDC fdc;
 
-void fdc_callback();
+void fdc_callback(void*);
 //#define SECTORS 9
 int lastbyte=0;
 uint8_t disc_3f7;
@@ -467,7 +467,7 @@ void fdc_write(uint16_t addr, uint8_t val, void *priv)
 //                                        fdc.stat = 0x10 | (fdc.stat & 0xf);
                                         discint = 8;
                                         fdc.pos = 0;
-                                        fdc_callback();
+                                        fdc_callback(NULL);
                                 }
                                 else
                                 {
@@ -498,13 +498,13 @@ void fdc_write(uint16_t addr, uint8_t val, void *priv)
                                 fdc.lastdrive = fdc.drive;
                                 discint = 0x0e;
                                 fdc.pos = 0;
-                                fdc_callback();
+                                fdc_callback(NULL);
                                 break;
                                 case 0x10: /*Get version*/
                                 fdc.lastdrive = fdc.drive;
                                 discint = 0x10;
                                 fdc.pos = 0;
-                                fdc_callback();
+                                fdc_callback(NULL);
                                 break;
                                 case 0x12: /*Set perpendicular mode*/
                                 fdc.pnum=0;
@@ -523,7 +523,7 @@ void fdc_write(uint16_t addr, uint8_t val, void *priv)
                                 fdc.lastdrive = fdc.drive;
                                 discint = fdc.command;
                                 fdc.pos = 0;
-                                fdc_callback();
+                                fdc_callback(NULL);
                                 break;
 
                                 case 0x18:
@@ -531,10 +531,10 @@ void fdc_write(uint16_t addr, uint8_t val, void *priv)
                                 fdc.lastdrive = fdc.drive;
                                 discint = 0x10;
                                 fdc.pos = 0;
-                                fdc_callback();
+                                fdc_callback(NULL);
                                 /* fdc.stat = 0x10;
                                 discint  = 0xfc;
-                                fdc_callback(); */
+                                fdc_callback(NULL); */
                                 break;
 
                                 default:
@@ -798,7 +798,7 @@ uint8_t fdc_read(uint16_t addr, void *priv)
         return temp;
 }
 
-void fdc_callback()
+void fdc_callback(void*)
 {
         int temp;
         int drive;

--- a/src/ide.c
+++ b/src/ide.c
@@ -1174,12 +1174,12 @@ abort_cmd:
 	ide_irq_raise(ide);
 }
 
-void ide_callback_pri()
+void ide_callback_pri(void*)
 {
 	callbackide(0);
 }
 
-void ide_callback_sec()
+void ide_callback_sec(void*)
 {
 	callbackide(1);
 }

--- a/src/keyboard.c
+++ b/src/keyboard.c
@@ -385,7 +385,7 @@ static uint8_t oldkey[272];
 static uint8_t keydelay[272];
 
 void (*keyboard_send)(uint8_t val);
-void (*keyboard_poll)();
+void (*keyboard_poll)(void*);
 int keyboard_scan = 1;
 
 static scancode *at_scancodes;

--- a/src/keyboard.h
+++ b/src/keyboard.h
@@ -1,5 +1,5 @@
 extern void (*keyboard_send)(uint8_t val);
-extern void (*keyboard_poll)();
+extern void (*keyboard_poll)(void*);
 void keyboard_process();
 extern int keyboard_scan;
 

--- a/src/keyboard_amstrad.c
+++ b/src/keyboard_amstrad.c
@@ -35,7 +35,7 @@ static int key_queue_start = 0, key_queue_end = 0;
 
 static uint8_t amstrad_systemstat_1, amstrad_systemstat_2;
 
-void keyboard_amstrad_poll()
+void keyboard_amstrad_poll(void*)
 {
         timer_advance_u64(&keyboard_amstrad.send_delay_timer, (1000 * TIMER_USEC));
         if (keyboard_amstrad.wantirq)

--- a/src/keyboard_amstrad.h
+++ b/src/keyboard_amstrad.h
@@ -1,3 +1,3 @@
 void keyboard_amstrad_init();
 void keyboard_amstrad_reset();
-void keyboard_amstrad_poll();
+void keyboard_amstrad_poll(void*);

--- a/src/keyboard_at.c
+++ b/src/keyboard_at.c
@@ -95,7 +95,7 @@ int mouse_queue_start = 0, mouse_queue_end = 0;
 
 void keyboard_at_adddata_keyboard(uint8_t val);
 
-void keyboard_at_poll()
+void keyboard_at_poll(void*)
 {
 	timer_advance_u64(&keyboard_at.send_delay_timer, (100 * TIMER_USEC));
 

--- a/src/keyboard_at.h
+++ b/src/keyboard_at.h
@@ -1,7 +1,7 @@
 void keyboard_at_init();
 void keyboard_at_init_ps2();
 void keyboard_at_reset();
-void keyboard_at_poll();
+void keyboard_at_poll(void*);
 void keyboard_at_set_mouse(void (*mouse_write)(uint8_t val, void *p), void *p);
 void keyboard_at_adddata_mouse(uint8_t val);
 

--- a/src/keyboard_olim24.c
+++ b/src/keyboard_olim24.c
@@ -43,7 +43,7 @@ static int key_queue_start = 0, key_queue_end = 0;
 
 static uint8_t mouse_scancodes[7];
 
-void keyboard_olim24_poll()
+void keyboard_olim24_poll(void*)
 {
         timer_advance_u64(&keyboard_olim24.send_delay_timer, (1000 * TIMER_USEC));
         //pclog("poll %i\n", keyboard_olim24.wantirq);

--- a/src/keyboard_olim24.h
+++ b/src/keyboard_olim24.h
@@ -1,5 +1,5 @@
 void keyboard_olim24_init();
 void keyboard_olim24_reset();
-void keyboard_olim24_poll();
+void keyboard_olim24_poll(void*);
 
 extern mouse_t mouse_olim24;

--- a/src/keyboard_pcjr.c
+++ b/src/keyboard_pcjr.c
@@ -41,7 +41,7 @@ struct
 static uint8_t key_queue[16];
 static int key_queue_start = 0, key_queue_end = 0;
 
-void keyboard_pcjr_poll()
+void keyboard_pcjr_poll(void*)
 {
         timer_advance_u64(&keyboard_pcjr.send_delay_timer, (220 * TIMER_USEC));
 

--- a/src/keyboard_pcjr.h
+++ b/src/keyboard_pcjr.h
@@ -1,3 +1,3 @@
 void keyboard_pcjr_init();
 void keyboard_pcjr_reset();
-void keyboard_pcjr_poll();
+void keyboard_pcjr_poll(void*);

--- a/src/keyboard_xt.c
+++ b/src/keyboard_xt.c
@@ -44,7 +44,7 @@ struct
 static uint8_t key_queue[16];
 static int key_queue_start = 0, key_queue_end = 0;
 
-void keyboard_xt_poll()
+void keyboard_xt_poll(void*)
 {
         timer_advance_u64(&keyboard_xt.send_delay_timer, (1000 * TIMER_USEC));
         if (!(keyboard_xt.pb & 0x40) && romset != ROM_TANDY)

--- a/src/keyboard_xt.h
+++ b/src/keyboard_xt.h
@@ -1,4 +1,4 @@
 void keyboard_xt_init();
 void keyboard_tandy_init();
 void keyboard_xt_reset();
-void keyboard_xt_poll();
+void keyboard_xt_poll(void*);

--- a/src/pit.h
+++ b/src/pit.h
@@ -1,7 +1,7 @@
 extern uint64_t PITCONST;
 void pit_init();
 void pit_ps2_init();
-void pit_reset();
+void pit_reset(PIT *pit);
 void pit_set_gate(PIT *pit, int channel, int gate);
 void pit_set_using_timer(PIT *pit, int t, int using_timer);
 void pit_set_out_func(PIT *pit, int t, void (*func)(int new_out, int old_out));

--- a/src/scsi_53c400.c
+++ b/src/scsi_53c400.c
@@ -740,7 +740,7 @@ static void *scsi_rt1000b_init()
         return scsi_53c400_init("Rancho_RT1000_RTBios_version_8.10R.bin");
 }
 
-static void *scsi_t130b_init(char *bios_fn)
+static void *scsi_t130b_init()
 {
         lcs6821n_t *scsi = malloc(sizeof(lcs6821n_t));
         memset(scsi, 0, sizeof(lcs6821n_t));

--- a/src/scsi_aha1540.c
+++ b/src/scsi_aha1540.c
@@ -2232,7 +2232,7 @@ static uint16_t port_sw_mapping[8] =
         0x330, 0x334, 0x230, 0x234, 0x130, 0x134, -1, -1
 };
 
-static void *scsi_aha1542c_init(char *bios_fn)
+static void *scsi_aha1542c_init()
 {
         aha154x_t *scsi = malloc(sizeof(aha154x_t));
         uint32_t addr;
@@ -2283,7 +2283,7 @@ static void *scsi_aha1542c_init(char *bios_fn)
         return scsi;
 }
 
-static void *scsi_bt545s_init(char *bios_fn)
+static void *scsi_bt545s_init()
 {
         aha154x_t *scsi = malloc(sizeof(aha154x_t));
         uint32_t addr;

--- a/src/sound.c
+++ b/src/sound.c
@@ -102,7 +102,7 @@ int sound_card_get_from_internal_name(char *s)
 	return 0;
 }
 
-void sound_card_init()
+void sound_card_init(int card)
 {
         if (sound_cards[sound_card_current].device)
                 device_add(sound_cards[sound_card_current].device);

--- a/src/sound.h
+++ b/src/sound.h
@@ -10,7 +10,7 @@ struct device_t *sound_card_getdevice(int card);
 int sound_card_has_config(int card);
 char *sound_card_get_internal_name(int card);
 int sound_card_get_from_internal_name(char *s);
-void sound_card_init();
+void sound_card_init(int card);
 void sound_set_cd_volume(unsigned int vol_l, unsigned int vol_r);
 
 #define CD_FREQ 44100

--- a/src/sound_adlibgold.c
+++ b/src/sound_adlibgold.c
@@ -125,7 +125,7 @@ static int treble_cut[6] =
         (int)(0.354 * 16384)  /*-3 dB - filter output is at +6 dB*/
 };
 
-void adgold_timer_poll();
+void adgold_timer_poll(void *p);
 void adgold_update(adgold_t *adgold);
 
 void adgold_update_irq_status(adgold_t *adgold)

--- a/src/vid_s3.c
+++ b/src/vid_s3.c
@@ -137,7 +137,7 @@ typedef struct s3_t
 #define INT_FIFO_EMP (1 << 3)
 #define INT_MASK     0xf
 
-void s3_updatemapping();
+void s3_updatemapping(s3_t *s3);
 
 void s3_accel_write(uint32_t addr, uint8_t val, void *p);
 void s3_accel_write_w(uint32_t addr, uint16_t val, void *p);

